corehq/ex-submodules/casexml/apps/phone/restore.py

```diff
@@ -381,7 +478,13 @@ def get_payload(self):
 
     def get_response(self):
         try:
-            return HttpResponse(self.get_payload(), mimetype="text/xml")
+            payload = self.get_payload()
+            try:
+                f = open(payload, 'r')
+                # Since payload file is all one line, need to readline based on bytes
+                return StreamingHttpResponse(iter(lambda: f.readline(MAX_BYTES), ''), mimetype="text/xml")
+            except IOError:
```

Comment:
ill check on the `ENOENT` seems promising. I'm not sure how that nested try catch would work. How does that differentiate between the two types of IO errors?